subprojects/docs/src/docs/userguide/jacocoPlugin.adoc

```diff
@@ -0,0 +1,220 @@
+// Copyright 2017 the original author or authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[jacoco_plugin]]
+== The JaCoCo Plugin
+
+
+[NOTE]
+====
+
+The JaCoCo plugin is currently <<feature_lifecycle,incubating>>. Please be aware that the DSL and other configuration may change in later Gradle versions.
+
+====
+
+The JaCoCo plugin provides code coverage metrics for Java code via integration with http://www.eclemma.org/jacoco/[JaCoCo].
+
+
+[[sec:jacoco_getting_started]]
+=== Getting Started
+
+To get started, apply the JaCoCo plugin to the project you want to calculate code coverage for.
+
+++++
+<sample id="applyJacoco" dir="testing/jacoco/quickstart" title="Applying the JaCoCo plugin">
+            <sourcefile file="build.gradle" snippet="apply-plugin"/>
+        </sample>
+++++
+
+If the Java plugin is also applied to your project, a new task named `jacocoTestReport` is created that depends on the `test` task. The report is available at `__$buildDir__/reports/jacoco/test`. By default, a HTML report is generated.
+
+[[sec:configuring_the_jacoco_plugin]]
+=== Configuring the JaCoCo Plugin
+
+The JaCoCo plugin adds a project extension named `jacoco` of type api:org.gradle.testing.jacoco.plugins.JacocoPluginExtension[], which allows configuring defaults for JaCoCo usage in your build.
+
+++++
+<sample id="configJacoco" dir="testing/jacoco/quickstart" title="Configuring JaCoCo plugin settings">
+            <sourcefile file="build.gradle" snippet="jacoco-configuration"/>
+        </sample>
+++++
+
+.Gradle defaults for JaCoCo properties
+[cols="a,a", options="header"]
+|===
+| Property
+| Gradle default
+
+| reportsDir
+| `__$buildDir__/reports/jacoco`
+|===
+
+
+[[sec:jacoco_report_configuration]]
+=== JaCoCo Report configuration
+
+The api:org.gradle.testing.jacoco.tasks.JacocoReport[] task can be used to generate code coverage reports in different formats. It implements the standard Gradle type api:org.gradle.api.reporting.Reporting[] and exposes a report container of type api:org.gradle.testing.jacoco.tasks.JacocoReportsContainer[].
```

Comment:
Is this correct? api:org.gradle.testing.jacoco.tasks.JacocoCoverageVerification[]